components/ai_chat/browser/ai_chat_tab_helper.cc

```diff
@@ -690,6 +695,22 @@ void AIChatTabHelper::MakeAPIRequestWithConversationHistoryUpdate(
                             std::move(data_received_callback));
 }
 
+void AIChatTabHelper::RetryAPIRequest() {
+  SetAPIError(mojom::APIError::None);
+  DCHECK(!chat_history_.empty());
+
+  std::vector<ConversationTurn>::reverse_iterator rit = chat_history_.rbegin();
+
+  for (; rit != chat_history_.rend(); ++rit) {
+    if ((*rit).character_type == CharacterType::HUMAN) {
+      auto turn = *std::make_move_iterator(rit);
+      chat_history_.erase(rit.base() - 1);
+      MakeAPIRequestWithConversationHistoryUpdate(turn);
+      break;
+    }
```

Comment:
Shouldn't we be removing any partial Assistant turns since the last Human turn too?
Also for this to work reliably we should really prevent new Human input whilst waiting for Assistant response, right?

Reply:
makes sense. It now removes the current turn and the one right after it